src/functions/uploadToRSIS/application/data-mapper/cat-cpc-mapper.ts

```diff
@@ -113,6 +116,19 @@ export const formatCPCTestCategory = (result: TestResultCatCPCSchema): string =>
   return result.category.substring(0, 1);
 };
 
-const formatCPCCombinationCode = (testData: TestData): string | null =>  {
+const formatCPCCombinationCode = (testData: TestData): string | null => {
   return testData.combination ? testData.combination.replace(/[^0-9]/g, '') : null;
 };
+
+/**
+ * Function to return null if DCPC otherwise return the first char of configuration from the vehicleDetails object
+ * Rigid - R, Articulated - A, Drawbar - D
+ * @param category
+ * @param vehicleDetails
+ */
+export const formatCPCVehicleDetails =
+  (category: CategoryCode, vehicleDetails: VehicleDetails | undefined): string | null => {
+    const configuration = get(vehicleDetails, 'configuration', null);
+    return category === TestCategory.CCPC ?
```

Comment:
as outcome of both elvis is the same (null), i think it should be more like this:

return category === TestCategory.CCPC && configuration ? configuration.substring(0, 1) : null;

Reply:
ammended

Reply:
ammended